lib/tasks/instrumentation_generator/instrumentation.thor

```diff
@@ -0,0 +1,95 @@
+# encoding: utf-8
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+require_relative '../../new_relic/language_support'
+require 'thor'
+
+class Instrumentation < Thor
+  include Thor::Actions
+
+  INSTRUMENTATION_ROOT = 'lib/new_relic/agent/instrumentation/'
+  MULTIVERSE_SUITE_ROOT = 'test/multiverse/suites/'
+  DEFAULT_SOURCE_LOCATION = 'lib/new_relic/agent/configuration/default_source.rb'
+
+  desc('scaffold NAME', 'Scaffold the required files for adding new instrumentation')
+  long_desc <<-LONGDESC
+    `instrumentation scaffold` requires one parameter by default: the name of the
+    library or class you are instrumenting. This task generates the basic
+    file structure needed to add new instrumentation to the Ruby agent.
+  LONGDESC
+
+  source_root(File.dirname(__FILE__))
+
+  option :method,
+    default: 'method_to_instrument',
+    desc: 'The method you would like to prepend or chain instrumentation onto'
+  option :args,
+    default: '*args',
+    desc: 'The arguments associated with the original method'
+
+  def scaffold(name)
+    @name = name
+    @method = options[:method] if options[:method]
+    @args = options[:args] if options[:args]
+    @class_name = ::NewRelic::LanguageSupport.camelize(name)
```

Comment:
@fallwith - your earlier comment about adding `@class_name` and creating a `camelize` method resolved by: 0150cc311ddc10051192bbb6a92a61af5a68d0bc and eea682478e8d1eb60efcf7e8bd53bd57b1c9c4c5